SimPEG/base/pde_simulation.py

```diff
@@ -232,10 +263,21 @@ def MfDeriv_prop(self, u, v=None, adjoint=False):
                 return Zero()
             stash_name = f"_Mf_{arg}_deriv"
             if getattr(self, stash_name, None) is None:
-                M_prop_deriv = self.mesh.get_face_inner_product_deriv(
-                    np.ones(self.mesh.n_cells)
-                )(np.ones(self.mesh.n_faces)) * getattr(self, f"{arg.lower()}Deriv")
-                setattr(self, stash_name, M_prop_deriv)
+                prop = getattr(self, arg.lower())
+                t_type = TensorType(self.mesh, prop)
+
+                M_func = self.mesh.get_face_inner_product_deriv(model=prop)
+                prop_deriv = getattr(self, f"{arg.lower()}Deriv")
+                if t_type < 3 and self.mesh._meshType.lower() in (
```

Comment:
a quick comment to refresh the memory on `t_type` would be helpful here 
```suggestion
                # t_type == 3 for full tensor, t_type < 3 for diagonal anisotropy, isotropic and scalar properties
                if t_type < 3 and self.mesh._meshType.lower() in (
```

Reply:
Sure thing, will be helpful for future iterations to remind ourselves.